### PR TITLE
Use different service accounts for each deployment

### PR DESCRIPTION
**Pull request by lookuptable**
This does not affect our current demo but makes it possible to later
demo service account based ACL using Istio auth.

---
This change is [Reviewable](https://reviewable.io/reviews/istio/istio/171)

### CODE REVIEW

File: kubernetes/istio-rbac/istio-rbac.yaml [hunk 1]
```diff
@@ -97,7 +97,22 @@ metadata:
   name: istio-sidecar-role-binding
 subjects:
 - kind: ServiceAccount
-  name: default
+  name: details-service-account
```
Comment (costinm) [inline kubernetes/istio-rbac/istio-rbac.yaml]: I don't think we can have the sample accounts in the 'default installation' instructions.

File: demos/apps/bookinfo/bookinfo.yaml [hunk 2]
```diff
@@ -63,6 +69,11 @@ spec:
   selector:
     app: ratings
 ---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: ratings-v1-service-account
```
Comment (costinm) [inline demos/apps/bookinfo/bookinfo.yaml]: Do we expect apps to have per-version service accounts ( v1 ) ? Seems to complex/fine grained.

Comment (costinm) [inline demos/apps/bookinfo/bookinfo.yaml]: I mean: typically different versions of an app would have the same permissions / roles, and I would expect
ACLs/permissions to be set to the role independent of the release.

File: kubernetes/istio-rbac/istio-rbac.yaml (hunk 1 above)
Comment (costinm) [inline kubernetes/istio-rbac/istio-rbac.yaml]: I mean: the service accounts and bindings are only specific to the bookinfo sample, should not be installed/used if istio is installed for a different app.

File: demos/apps/bookinfo/bookinfo.yaml (hunk 2 above)
Comment (rshriram) [inline demos/apps/bookinfo/bookinfo.yaml]: I agree. Lets keep the service accounts independent of the versions.

File: kubernetes/istio-rbac/istio-rbac.yaml (hunk 1 above)
Comment (myidpt) [inline kubernetes/istio-rbac/istio-rbac.yaml]: I agree that we should make istio-auth-X.yaml and istio-rbac.yaml generic configurations, not specific to bookinfo. We are referring to these files in our general tutorial pages.
I think setting service accounts and bindings for bookinfo applications shall be moved out to bookinfo.yaml, if possible.

**Comment by rshriram** [comment]
whats the resolution on this?

**Comment by rshriram** [comment]
closing until folks start looking at it again.

**Pull request closed without merging**